From 08084b1179bb9fc38bc22b464b3d44907107bfd3 Mon Sep 17 00:00:00 2001
From: Simo Sorce <simo@redhat.com>
Date: Tue, 25 Apr 2017 12:39:32 +0000
Subject: [PATCH 133/135] ssh tools: The ai structure is not an array,
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This structure is actually a linked list, so do not mislead readers by
treating it as an array.

Resolves:
https://pagure.io/SSSD/sssd/issue/1498

Merges: https://pagure.io/SSSD/sssd/pull-request/3383

Signed-off-by: Simo Sorce <simo@redhat.com>
Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
---
 src/sss_client/ssh/sss_ssh_knownhostsproxy.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/sss_client/ssh/sss_ssh_knownhostsproxy.c b/src/sss_client/ssh/sss_ssh_knownhostsproxy.c
index adb82288d435cefccf7e23e6ed2b2c551798a7f8..310243c2fc8091f711559d4afb412e619af687ad 100644
--- a/src/sss_client/ssh/sss_ssh_knownhostsproxy.c
+++ b/src/sss_client/ssh/sss_ssh_knownhostsproxy.c
@@ -268,10 +268,10 @@ int main(int argc, const char **argv)
             DEBUG(SSSDBG_OP_FAILURE,
                   "getaddrinfo() failed (%d): %s\n", ret, gai_strerror(ret));
         } else {
-            host = ai[0].ai_canonname;
+            host = ai->ai_canonname;
         }
     } else {
-        ret = getnameinfo(ai[0].ai_addr, ai[0].ai_addrlen,
+        ret = getnameinfo(ai->ai_addr, ai->ai_addrlen,
                           canonhost, NI_MAXHOST, NULL, 0, NI_NAMEREQD);
         if (ret) {
             DEBUG(SSSDBG_OP_FAILURE,
@@ -295,7 +295,7 @@ int main(int argc, const char **argv)
     if (pc_args) {
         ret = connect_proxy_command(discard_const(pc_args));
     } else if (ai) {
-        ret = connect_socket(ai[0].ai_family, ai[0].ai_addr, ai[0].ai_addrlen);
+        ret = connect_socket(ai->ai_family, ai->ai_addr, ai->ai_addrlen);
     } else {
         ret = EFAULT;
     }
-- 
2.12.2

